crates/storage/provider/src/providers/consistent.rs

```diff
@@ -797,20 +797,32 @@ impl<N: ProviderNodeTypes> BlockReader for ConsistentProvider<N> {
         source: BlockSource,
     ) -> ProviderResult<Option<Self::Block>> {
         match source {
```

Comment:
this match requires us to duplicate both arms, we can simplify this if we use two matches!(Canonical|Any) matches!(Pending|Any) and only return if some

and fallback is None